numpyro/distributions/directional.py

```diff
@@ -398,14 +398,12 @@ def norm_const(self):
         den = special.gammaln(m + 1.0)
         lbinoms = num - 2 * den
 
-        fs = (
-            lbinoms.reshape(-1, 1)
-            + 2 * m * jnp.log(corr)
-            - m * jnp.log(4 * jnp.prod(conc, axis=-1))
+        fs = lbinoms.reshape(-1, 1) + m * (
+            jnp.log(jnp.clip(corr**2, a_min=jnp.finfo(float).tiny))
```

Comment:
The float finfo is too small for single precision, how about `jnp.finfo(jnp.result_type(float)).tiny`?